src/test/java/com/arpnetworking/tsdcore/sinks/KairosDbSinkTest.java

```diff
@@ -146,6 +148,27 @@ public void testPost() throws InterruptedException, IOException {
         Mockito.verify(_mockMetrics, Mockito.times(1)).close();
     }
 
+     @Test
+    public void testPostFailure() throws InterruptedException, IOException {
+         // Fake a successful post to KairosDb
+        _wireMock.register(WireMock.post(WireMock.urlEqualTo(PATH))
+                .willReturn(WireMock.aResponse()
+                        .withStatus(404)));
+        _kairosDbSinkBuilder.setMaxRetries(1).build().recordAggregateData(TestBeanFactory.createPeriodicData());
+        Thread.sleep(3000);
```

Comment:
(I recognize there's already a Thread.sleep(1000) in this test case, but - why 3000?)

Reply:
Because I added exponential backoff for the retry, which means, after initial HTTPrequest, the thread will wait 1000 milliseconds before sending the next attempt. So if we use 1000 here this test will fail.

Reply:
Sleeps are a really bad way to deal with this and will cause brittle tests. Waiting for an asyc completion with a timeout is probably a better way to go